syntax = "proto3";

package stt;

// The STT service definition
service STTService {
  rpc SendSTTResult(STTResult) returns (STTResponse);
  rpc ProcessAudioFile(AudioFileRequest) returns (STTResponse);
}

message STTResult {
  string user_id = 1;
  string transcript = 2;
  string language = 3;
}

message AudioFileRequest {
  string user_id = 1;
  string audio_url = 2;
  string language = 3;
}

message STTResponse {
  string status = 1;
  string message = 2;
}
